importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -3030,6 +3081,65 @@ MusicXMLDelayedDirectionElement* MusicXMLInferredFingering::toDelayedDirection()
       return dd;
       }
 
+//---------------------------------------------------------
+//   convertTextToNotes
+//---------------------------------------------------------
+/**
+ Converts note characters in _wordsText to proper symbols and
+ returns the tempo value of the resulting notes
+ */
+
+double MusicXMLParserDirection::convertTextToNotes()
+      {
+      QRegularExpression notesRegex("(?<note>[yxeqhwW]\\.{0,2})(\\s*=)");
+      QString notesSubstring = notesRegex.match(_wordsText).captured("note");
+
+      QList<QPair<QString, QString>> noteSyms{{"q", QString("<sym>metNoteQuarterUp</sym>")},   // note4_Sym
+                                                {"e", QString("<sym>metNote8thUp</sym>")},       // note8_Sym
+                                                {"h", QString("<sym>metNoteHalfUp</sym>")},      // note2_Sym
+                                                {"y", QString("<sym>metNote32ndUp</sym>")},      // note32_Sym
+                                                {"x", QString("<sym>metNote16thUp</sym>")},      // note16_Sym
+                                                {"w", QString("<sym>metNoteWhole</sym>")},
+                                                {"W", QString("<sym>metNoteDoubleWhole</sym>")}};
+      for (auto noteSym : noteSyms) {
+            if (notesSubstring.contains(noteSym.first)) {
+                  notesSubstring.replace(noteSym.first, noteSym.second);
+                  break;
+                  }
+            }
+      notesSubstring.replace(".", QString("<sym>metAugmentationDot</sym>")); // dot
+      _wordsText.replace(notesRegex, notesSubstring + "\\2");
+
+      double tempoValue = TempoText::findTempoValue(_wordsText);
+      if (!tempoValue) tempoValue = 1.0 / 60.0; // default to quarter note
+      return tempoValue;
+      }
+
+//---------------------------------------------------------
+//   attemptTempoTextCoercion
+//---------------------------------------------------------
+/**
+ Infers if a direction is likely tempo text, possibly changing
+ the _wordsText to the appropriate note symbol and inferring the _tpoSound.
+ */
+
+bool MusicXMLParserDirection::attemptTempoTextCoercion(const Fraction& tick)
+      {
+      QList<QString> tempoWords{"rit", "rall", "accel", "tempo", "allegr", "poco", "molto", "più", "meno", "mosso", "rubato"};
```

Comment:
There's at least few terms in there that might not be tempo at all (e.g. poco _**f**_), but it seems a kinda arbitrary list, what percentage of cases do you expect it to cover?  Why "allegr" but not, say, "moderato", "andante" or "lento"?
And obviously a lot of music these days doesn't use Italian directions at all...but I guess I'm not fully clued in with what you're trying to solve here.

Reply:
Yes, at this point it's certainly non-exhaustive list. We currently have some XML scores that we are testing with, so these are pulled from there. The list can certainly grow over time.

As far as the words that may be part of things other than tempo markings (poco, molto), note that this list is only used _after_ it is checked that the text is above the staff and bold (I'll also use the slightly-more-robust regex version of bold-checking I mention above). It would be very rare (and likely incorrect) to have a fully-bold direction above the staff with the word poco that _isn't_ tempo text.

That said, there was a missing bracket in the code that uses this list. Glad you said something!

Reply:
I think you should at least support "Egyptically": https://pbs.twimg.com/media/D1H2tmKU8AAtK2w?format=jpg&name=small